tests/test_scheduler.py

```diff
@@ -34,6 +34,7 @@ def test_scheduler_starts_workflows(
     with patch.multiple(
         "reana_server.scheduler",
         reana_ready=Mock(return_value=True),
+        check_memory_availability=Mock(return_value=True),
```

Comment:
Would it be feasible to add some tests for the new functionality? Also for the complexity estimator would be handy.